SCClassLibrary/Common/Streams/Stream.sc

```diff
@@ -359,8 +359,12 @@ PauseStream : Stream {
 	}
 	reset { originalStream.reset }
 	stop {
-		this.prStop;
```

Comment:
One possible concern with this: prior to the change, it would call `prStop` and then broadcast the `\userStopped` notification. Now, if the task is stopped from outside, it broadcasts the notification and then calls `prStop`.

Because it's a notification, we have no idea what kind of function is registered. The user's response to the notification might assume that `stream == nil` (as set by `prStop`)... but we are now breaking that assumption, and I don't see any good reason to break it, except for the convenience of handling the stream variable. That's not a good enough reason.

I would prefer this:

```
	stop {
		var saveStream = this.stream;
		this.prStop;
 		this.changed(\userStopped);
		if(saveStream === thisThread) {
			nil.alwaysYield
		}
	}
```

Reply:
great, I had already worried - this is much better.